zeroconf.py

```diff
@@ -971,12 +971,12 @@ def __init__(self, zc, type_, handlers=None, listener=None):
         self.delay = _BROWSER_TIME
         self._handlers_to_call = []
 
+        self._service_state_changed = Signal()
+
         self.done = False
 
         self.zc.add_listener(self, DNSQuestion(self.type, _TYPE_PTR, _CLASS_IN))
```

Comment:
A likely better change to address this is to instead move the offending:

```python

self.zc.add_listener(self, DNSQuestion(self.type, _TYPE_PTR, _CLASS_IN))

```

To the first line in the ```ServiceBrowser.run()``` method.  This would eliminate all race conditions as the ```ServiceBrowser()``` would be fully instantiated before opening for business..